### DISCUSSION TITLE
Compute water balance from MIKE SHE outputs

### DISCUSSION THREAD

**Discussion by MilanFischer** [DHI/mikeio#582] [Q&A]
Hello, does anybody have an experience with computing the water balance from MIKE SHE dfs2/dfs3 results file outside of the wbl.exe utility? My main motivation to do so is to parallelize the water balance computation.

Thanks,
Milan

**Comment by jsmariegaard** [comment]
I have no MIKE SHE experience, @watermain would you know this?

**Comment by watermain** [comment]
Hi Milan, I have not heard of anyone trying this, and I´m a bit doubtful it is practical. It depends a bit on the complexity of the water balance you have in mind, if it is just a pretty simple subset then it might work out. For a complete water balance I believe it would be a lot of work, including a lot of trial and error and testing against the water balance tool to get close. If at all possible I´d go with the provided tool - maybe you can simplify the wbl setup so it runs faster.

**Comment by MilanFischer** [comment]
Hi Jesper and Uwe,
thank you both for your response.  I honestly naively expected that the water balance would be pretty straightforward balancing several layers which are the outputs from MIKE SHE. However, I understand your concern. In that case,  I will try to play around with the simplification of the wbl setup. As I mentioned, my main motivation to do it outside of the wbl tool was a hope for easier parallelization. My experience is that if I run several wbl tool at once, their performance goes significantly down (even at machine with large CPU/RAM resources) but the reason form me remains unclear.

Cheers,
Milan

**Comment by watermain** [comment]
Usually running many separate processes scales a lot better than parallel threads in a single process. A few ideas where  to look further: Is cpu the limiting factor or does disk access also play a role? Are you running multiple wbl postprocessings accessing the same gross files - then maybe it could help to make copies before? Does it help to switch off hyperthreading? The operating system distributes the load between different cpus, sometimes it can happen that the distribution is not optimal. The windows task manager might already give you a rough idea if all cpus are evenly loaded. For more details check out cpu affinity (https://en.wikipedia.org/wiki/Processor_affinity).
